hols/todo/MultiCluster.md

```diff
@@ -0,0 +1,357 @@
+```
+clusters = {
+  c1 = { region = "paris", vcn = "10.1.0.0/16", pods = "10.201.0.0/16", services = "10.101.0.0/16", enabled = true }
+  c2 = { region = "marseille", vcn = "10.2.0.0/16", pods = "10.202.0.0/16", services = "10.102.0.0/16", enabled = true }
+  # c3 = { region = "frankfurt", vcn = "10.3.0.0/16", pods = "10.203.0.0/16", services = "10.103.0.0/16", enabled = true }  
+}
+    
+}
+```
+4. Run terraform init and apply:
+
+```
+terraform init
+terraform apply --auto-approve
+```
+5. RPCs will have been already created. Login to OCI Console and peer the VCNs using the RPCs.
+
+### Configure Multi cluster
+1. ssh to the operator host
+
+2. Deploy Cilium in all clusters:
+```bash
+for c in c1 c2; do
+  kubectx $c
+  helm install cilium cilium/cilium --namespace=kube-system -f $HOME/cilium/cilium-$c.yaml
+done
+```
+
+3. Delete pods not managed by Cilium:
+
+```
+for c in c1 c2; do
+  kubectx $c
+  bash $HOME/cilium_delete_pods.sh
+done
+
+for c in c1 c2; do
+  kubectx $c
+  cilium status
+done
+```
+
+4. Connect the clusters:
+
+```bash
+cilium clustermesh connect --context c1 --destination-context c2
+```
+> Note: If you have more than 2 clusters, connect all the clusters to each other in a mesh:
+> ```
+>   cilium clustermesh connect --context c1 --destination-context c3
+>   cilium clustermesh connect --context c2 --destination-context c3
+>   ```
+5. Check multi-cluster status:
+
+```
+for c in c1 c2; do
+  kubectx $c
+  cilium clustermesh status
+done
+```
+6. Expose CoreDNS for each cluster as a Network Load Balancer:
+
+```bash
+for c in c1 c2; do
+  kubectx $c
+  kubectl apply -f coredns/kubedns-$c.yaml
+done
+```
+
+6. Obtain the IP addresses of NLBs:
+
+```bash
+for c in c1 c2; do
+  kubectx $c
+  kubectl -n kube-system get svc kube-dns-lb
+done
+```
+7. Edit the CoreDNS configuration files in $HOME/coredns/ for c1, and respectively and replace each NLB's private IP address.
+
+8. Create the CoreDNS ConfigMap and restart CoreDNS for each cluster:
+
+```bash
+for c in c1 c2; do
+  kubectx $c
+  kubectl apply -f coredns/coredns-$c.yaml
+  kubectl delete pod --namespace kube-system -l k8s-app=kube-dns  
+done
+```
+9. Install Istio in each cluster:
+
+```bash
+bash $HOME/istio/install_istio.sh
+```
+
+
+## B. Database
+
+### Configure DR for Autonomous Database
+1. In OCI Console, navigate to the Autonomous Database instance page
+2. Under Resources, select Disaster Recovery
+3. Click on 'Add Peer Database'
+4. Select an OCI Region
+5. Select Autonomous Data Guard as the disaster recovery type
+6. Check 'Enable cross-region backup replication to disaster recovery peer'
+7. Click on 'Add Peer Database' to finish creating the Autonomous DR instance
+
+## C. Kubernetes Operators
+
+We will use 2 Kubernetes Operators:
+
+1. Oracle Coherence Operator: this will be used to deploy the Coherence cluster to OKE.
+2. TODO: External Secrets Operator: this will be used to retrieve the Hibernate Configuration and Wallet Secret from OCI Vault.
```

Comment:
TODO comment.

Reply:
If you want to state that there is a gap that can be addressed later make it very explicit with a separate paragraph and a link to an issue.

Reply:
done